src/AppBundle/View/I18nResponseBuilder.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/*
+ * This file is part of the Sententiaregum project.
+ *
+ * (c) Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ * (c) Ben Bieler <benjaminbieler2014@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace AppBundle\View;
+
+use Symfony\Component\Translation\TranslatorInterface;
+use Symfony\Component\Validator\ConstraintViolation;
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+/**
+ * Concrete implementation of a builder which transforms constraint violations into an i18n response.
+ *
+ * It converts a violation list to the following format:
+ *
+ * [
+ *   'de' => ['German translation'],
+ *   'en' => ['English translation'],
+ * ]
+ *
+ * Furthermore it is possible to sort all the violations by the property which caused the violation:
+ *
+ * [
+ *   'property' => [
+ *     [
+ *       'de' => ['German translation'],
+ *       'en' => ['English translation'],
+ *     ]
+ *   ],
+ *   'another_property' => [
+ *     [
+ *       // ...
+ *     ]
+ *   ]
+ * ]
+ *
+ * @author Maximilian Bosch <maximilian.bosch.27@gmail.com>
+ */
+class I18nResponseBuilder implements I18nResponseBuilderInterface
+{
+    /**
+     * @var TranslatorInterface
+     */
+    private $translator;
+
+    /**
+     * Constructor.
+     *
+     * @param TranslatorInterface $translator
+     */
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildTranslatableResponse(
+        ConstraintViolationListInterface $violations,
+        $sortProperties = true,
+        $useAllLanguages = true,
+        array $targetLocales = [],
+        $domain = 'messages'
+    ) {
+        $violationList = iterator_to_array($violations, false);
+        $hasLocales    = count($targetLocales) > 0;
+
+        if (!$useAllLanguages && $hasLocales || $useAllLanguages && !$hasLocales) {
+            throw new \InvalidArgumentException(
+                'Wrong usage of $targetLocales: If all languages should be rendered, $targetLocales should be set, if not, $targetLocales should be an empty array!'
+            );
+        }
+
+        return array_reduce(
```

Comment:
I'm still not sure whether ``array_walk`` is better or an imperative foreach approach